ruby/ql/lib/codeql/ruby/security/LdapInjectionCustomizations.qll

```diff
@@ -0,0 +1,68 @@
+/**
+ * Provides default sources, sinks and sanitizers for detecting
+ * LDAP Injections, as well as extension points for adding your own
+ */
+
+private import codeql.ruby.Concepts
+private import codeql.ruby.DataFlow
+private import codeql.ruby.dataflow.BarrierGuards
+private import codeql.ruby.dataflow.RemoteFlowSources
+private import codeql.ruby.ApiGraphs
+
+/**
+ * Provides default sources, sinks and sanitizers for detecting
+ * LDAP Injections, as well as extension points for adding your own
+ */
+module LdapInjection {
+  /** A data flow source for LDAP Injection vulnerabilities */
+  abstract class Source extends DataFlow::Node { }
+
+  /** A data flow sink for LDAP Injection vulnerabilities */
+  abstract class Sink extends DataFlow::Node { }
+
+  /** A sanitizer for LDAP Injection vulnerabilities. */
+  abstract class Sanitizer extends DataFlow::Node { }
+
+  /**
+   * Additional taint steps for "LDAP Injection" vulnerabilities.
+   */
+  predicate isAdditionalTaintStep(DataFlow::Node nodeFrom, DataFlow::Node nodeTo) {
+    filterTaintStep(nodeFrom, nodeTo)
+  }
+
+  private predicate filterTaintStep(DataFlow::Node n1, DataFlow::Node n2) {
+    exists(DataFlow::CallNode filterCall |
+      (
+        filterCall =
+          API::getTopLevelMember("Net").getMember("LDAP").getMember("Filter").getAMethodCall(["eq"]) or
+        filterCall.getMethodName() = ["[]"]
```

Comment:
I guess that this is for the `attributes` keyword arg in a case like: 
```rb 
ldap.search(base: "ou=people,dc=example,dc=com", filter: "cn=George", attributes: [name])
```
where we want to consider that a tainted element taints the whole array?

If so, we should maybe rename this predicate as `attributeArrayTaintStep` and make it such that it propagates flow from any indexed argument (`getArgument(_)`). It's also worth adding a line of documenting why we need this flow step.